In [1]:
import os
import tensorflow as tf
import numpy as np

In [2]:
NUM_CLASSES = 11

In [3]:
datagen = tf.keras.preprocessing.image.ImageDataGenerator(
    preprocessing_function=tf.keras.applications.vgg19.preprocess_input,
    rescale=1.0/255.0
)

In [4]:
def evaluate_model(model_path, target_dim, batch_size):
    
    test_generator = datagen.flow_from_directory(
        directory='./evaluation/', 
        target_size=(target_dim, target_dim),
        batch_size=batch_size,
        shuffle=False
    )
    
    model = tf.keras.models.load_model(model_path)
    return model.evaluate_generator(test_generator, steps=test_generator.samples // batch_size)

In [5]:
models_path = os.listdir('./models/')

In [6]:
models_path

['.ipynb_checkpoints',
 'inceptionV3-06-0.89.hdf5',
 'inceptionV3-regularized-03-0.82.hdf5',
 'inceptionV3-regularized-04-0.83.hdf5',
 'mobilenet-05-0.91.hdf5',
 'mobilenet-regularized-03-0.78.hdf5',
 'mobilenet-regularized-05-0.80.hdf5',
 'vgg19-34-0.90.hdf5',
 'vgg19-regularized-08-0.78.hdf5',
 'vgg19-regularized-18-0.78.hdf5']

In [7]:
evaluations = []

In [10]:
## Inception V3 
for path in models_path[1:4]:
    evaluation = evaluate_model("./models/" + path, 300, 128)
    print("Model: " + path)
    print(evaluation)
    evaluations.append(evaluation)

Found 3347 images belonging to 11 classes.
Model: inceptionV3-06-0.89.hdf5
[0.6018862128257751, 0.8109976053237915]
Found 3347 images belonging to 11 classes.
Model: inceptionV3-regularized-03-0.82.hdf5
[0.8373874425888062, 0.854567289352417]
Found 3347 images belonging to 11 classes.
Model: inceptionV3-regularized-04-0.83.hdf5
[0.783935546875, 0.8530648946762085]


### Result

inceptionV3-regularized-04-0.83.hdf5 model yielded accuracy of 85% with 0.783 loss.

In [20]:
for path in models_path[4:7]:
    evaluation = evaluate_model("./models/" + path, 224, 128)
    print("Model: " + path)
    print(evaluation)
    evaluations.append(evaluation)

Found 3347 images belonging to 11 classes.
Model: mobilenet-05-0.91.hdf5
[0.6709072589874268, 0.8013821840286255]
Found 3347 images belonging to 11 classes.
Model: mobilenet-regularized-03-0.78.hdf5
[1.20906400680542, 0.8323317170143127]
Found 3347 images belonging to 11 classes.
Model: mobilenet-regularized-05-0.80.hdf5
[0.8465911149978638, 0.83203125]


### Result

mobilenet-regularized-05-0.80.hdf5 yielded accuracy of 83% with 0.846 loss.

In [9]:
for path in models_path[7:]:
    evaluation = evaluate_model("./models/" + path, 300, 32)
    print("Model: " + path)
    print(evaluation)
    evaluations.append(evaluation)

Found 3347 images belonging to 11 classes.
Instructions for updating:
Please use Model.evaluate, which supports generators.
Model: vgg19-34-0.90.hdf5
[0.8947008848190308, 0.7806490659713745]
Found 3347 images belonging to 11 classes.
Model: vgg19-regularized-08-0.78.hdf5
[0.731843888759613, 0.8377403616905212]
Found 3347 images belonging to 11 classes.
Model: vgg19-regularized-18-0.78.hdf5
[0.7413175106048584, 0.807692289352417]


### Result

vgg19-regularized-08-0.78.hdf5 model yielded 83.7% of accuracy with 0.731 loss.

## Conclusion

From mobilenet, vgg19, inceptionV3 models, inceptionV3 is doing well on test data.